README
Task-Template-Benchmark

This distribution contains no actual code, it simply exists to provide
a list of dependencies to assist in quickly installing all template engines
and optional dependencies used by Template::Benchmark.

Be warned, between them, the 34+ modules installed by this task have a
huge number of dependencies and prerequisites, on a fresh build of Perl
with only Bundle::CPAN installed this will run for over 20 minutes.

The current release of Task::Template::Benchmark aims to always track
the dependencies required by the current release of Template::Benchmark.

However, it should also be perfectly safe to use the current
Task::Template::Benchmark with an older install of Template::Benchmark
- you'll most likely only end up installing some extra modules that won't
be used by that older version.

INSTALLATION

To install this module, run the following commands:

  perl Build.PL
  ./Build installdeps
  ./Build test
  ./Build install

SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the
perldoc command.

    perldoc Task::Template::Benchmark

You can also look for information at:

    RT, CPAN's request tracker
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Task-Template-Benchmark

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/Task-Template-Benchmark

    CPAN Ratings
        http://cpanratings.perl.org/d/Task-Template-Benchmark

    Search CPAN
        http://search.cpan.org/dist/Task-Template-Benchmark/


LICENSE AND COPYRIGHT

Copyright (C) 2010 Sam Graham

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.



Changes
Revision history for Task-Template-Benchmark

1.00_06 Version released at .  *** Development version. ***
        Added requirements:
            Template::AutoFilter.
            Template::Simple.
            Text::Caml.
            Text::FastTemplate.
            Text::MiniTmpl.
            Text::ScriptTemplate.
            Text::SimpleTemplate.
            Text::TemplateLite.
            Text::Templet.
            Text::Xslate::Syntax::HTMLTemplate.

1.00_05 Version released at 2010-10-19-08:47.  *** Development version. ***
        Require latest Text::Xslate and Data::MessagePack for Windows compat.
        Text::Xslate family of engines no-longer on Windows blacklist.

1.00_04 Version released at 2010-10-18-10:32.  *** Development version. ***
        Support for Text::Clevy rename to Text::Clevery.

1.00_03 Version released at 2010-10-04-19:32.  *** Development version. ***
        Added HTML::Mason requirement.
        Added Solution requirement.
        Added HTML::Macro requirement.
        Added Cache::Ref requirement.

1.00_02 Version released at 2010-08-15-22:30.  *** Development version. ***
        Added Parse::Template requirement.

1.00_01 Version released at 2010-08-12-11:37.  *** Development version. ***
        Updated version requirements for TextXslateTT plugin.
        Required version 0.36 of Module::Build for "installdeps".

1.00    Version released at 2010-08-02-06:58.
        Added "see also" to POD.
        Stripped POD from README intro.
        Typo fixes in documentation.

0.99_01 Version released at 2010-07-27-16:48.  *** Development version. ***
        Initial release.


lib/Task/Template/Benchmark.pm
package Task::Template::Benchmark;

use warnings;
use strict;

our $VERSION = '1.00_05';

1;

__END__

=pod

=head1 NAME

Task::Template::Benchmark - Task to install all template engines benchmarked by Template::Benchmark.

=head1 SYNOPSIS

This distribution contains no actual code, it simply exists to provide
a list of dependencies to assist in quickly installing all template engines
and optional dependencies used by L<Template::Benchmark>.

Be warned, between them, the 34 modules installed by this task have a
huge number of dependencies and prerequisites, on a fresh build of Perl
with only Bundle::CPAN installed this will run for over 20 minutes.

To use L<Task::Template::Benchmark> simply type at your CPAN prompt:

  install Task::Template::Benchmark

Or extract the distribution tarball to a directory and do the following:

  perl Build.PL
  ./Build installdeps
  ./Build test
  ./Build install

The current release of L<Task::Template::Benchmark> aims to always track
the dependencies required by the current release of L<Template::Benchmark>.

However, it should also be perfectly safe to use the current
L<Task::Template::Benchmark> with an older install of L<Template::Benchmark>
- you'll most likely only end up installing some extra modules that won't
be used by that older version.

=head1 INCLUDED MODULES

=over

=item Template Benchmark itself:

L<Template:Benchmark>

=item Optional bits for extra L<benchmark_template_engines> behaviour:

L<Term::ProgressBar::Simple>

L<JSON::Any>

=item Modules required by multiple plugins:

L<File::Spec>

L<File::Spec> needed by plugins for:
L<Mojo::Template>, L<Tenjin>, L<Text::MicroMason>, L<Text::Tmpl>.

L<IO::File>

L<IO::File> needed by plugins for:
L<Mojo::Template>, L<Tenjin>.

=item Template Engines:

L<HTML::Mason>

L<HTML::Macro>

L<HTML::Template>

L<HTML::Template::Compiled>

L<HTML::Template::Expr>

L<HTML::Template::JIT>

L<HTML::Template::Pro>

L<Mojo>

L<Mojo::Template>

L<NTS::Template>

L<Solution> 0.000000004 (for sorted hash-looping)

L<Template::Alloy>

L<Template::AutoFilter>

L<Template::Sandbox>

Extras for L<Template::Sandbox>:

L<Cache::CacheFactory>
L<Cache::Cache>
L<Cache::FastMemoryCache>
L<Cache::FastMmap>
L<Cache::Ref>
L<CHI>

L<Template::Simple>

L<Template::Tiny>

L<Template>

Extras for L<Template>:

L<Template::Stash::XS>
L<Template::Parser::CET>

L<Tenjin> 0.05 (pre-0.05 Tenjin was an incompatible API change)

L<Text::Caml>

L<Text::ClearSilver>

L<Text::Clevery>

L<Text::MicroMason>

L<Text::MicroTemplate>

L<Text::MicroTemplate::Extended>

L<Text::MiniTmpl>

L<Text::ScriptTemplate>

L<Text::SimpleTemplate>

L<Text::Templet>

L<Text::Template>

L<Text::Template::Simple>

L<Text::Tmpl>

L<Text::Xslate> 0.1053 (0.1053 required for TT2 bridge support)

L<Text::Xslate::Bridge::TT2>

=back

=head1 WINDOWS SUPPORT

Under Windows the following modules are not installed because they
appear to fail on Windows:

=over

=item L<HTML::Template::Compiled>

=item L<HTML::Template::JIT>

=item L<Text::ClearSilver>

=item L<Term::ProgressBar::Simple>

These modules appear to have build failures under windows, or prerequisites
that fail to build.

=item L<NTS::Template>

Returns empty content.

=item L<Template::Alloy>

The L<HTML::Template> emulation for L<Template::Alloy> appears to
get confused by the volume letters in Windows filenames.

=item L<Template::Tiny>

=item L<Text::Template::Simple>

Error on attempting to run the template.

=item L<Text::Tmpl>

Returns corrupted output at tail end of template.

=back

=head1 KNOWN ISSUES AND BUGS

None currently known.

=head1 SEE ALSO

L<Template::Benchmark>

=head1 AUTHOR

Sam Graham, C<< <libtemplate-benchmark-perl BLAHBLAH illusori.co.uk> >>

=head1 BUGS

Please report any bugs or feature requests to C<bug-task-template-benchmark at rt.cpan.org>, or through
the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Task-Template-Benchmark>.  I will be notified, and then you'll
automatically be notified of progress on your bug as I make changes.




=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Task::Template::Benchmark


You can also look for information at:

=over 4

=item * RT: CPAN's request tracker

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Task-Template-Benchmark>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Task-Template-Benchmark>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Task-Template-Benchmark>

=item * Search CPAN

L<http://search.cpan.org/dist/Task-Template-Benchmark/>

=back


=head1 LICENSE AND COPYRIGHT

Copyright 2010 Sam Graham.

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.

=cut


Build.PL
use strict;
use warnings;
use Module::Build;

my %modules = (
        #  Template::Benchmark itself.
        'Template::Benchmark'       => 0,

        #  Optional bits for extra benchmark_template_engines behaviour.
        'Term::ProgressBar::Simple' => 0,
        'JSON::Any'                 => 0,

        #  File::Spec needed by plugins for:
        #    Mojo::Template, Tenjin, Text::MicroMason, Text::Templ.
        'File::Spec'                => 0,

        #  IO::File needed by plugins for:
        #    Mojo::Template, Tenjin.
        'IO::File'                  => 0,

        'HTML::Mason'               => 0,
        'HTML::Macro'               => 0,
        'HTML::Template'            => 0,
        'HTML::Template::Compiled'  => 0,
        'HTML::Template::Expr'      => 0,
        'HTML::Template::JIT'       => 0,
        'HTML::Template::Pro'       => 0,
        'Mojo'                      => 0,
        'Mojo::Template'            => 0,
        'NTS::Template'             => 0,
        'Parse::Template'           => 0,
        #  Need 0.000000004 for sorted hash-looping.
        'Solution'                  => '0.000000004',
        'Template::Alloy'           => 0,
        #  Lots of extras here...
        'Template::Sandbox'         => 0,
        'Cache::CacheFactory'       => 0,
        'Cache::Cache'              => 0,
        'Cache::FastMemoryCache'    => 0,
        'Cache::FastMmap'           => 0,
        'Cache::Ref'                => 0,
        'CHI'                       => 0,
        'Template::AutoFilter'      => '0.130420',
        'Template::Tiny'            => 0,
        'Template'                  => 0,
        'Template::Simple'          => '0.06',
        'Template::Stash::XS'       => 0,
        'Template::Parser::CET'     => 0,
        #  pre-0.05 Tenjin was an incompatible API change
        'Tenjin'                    => '0.05',
        'Text::Caml'                => '0.10',
        'Text::ClearSilver'         => 0,
        'Text::Clevery'             => 0,
        'Text::FastTemplate'        => '0.95',
        'Text::MicroMason'          => 0,
        'Text::MicroTemplate'       => 0,
        'Text::MicroTemplate::Extended' => 0,
        'Text::MiniTmpl'            => '1.1.1',
        'Text::ScriptTemplate'      => '0.08',
        'Text::SimpleTemplate'      => '0.36',
        'Text::Template'            => 0,
        'Text::Template::Simple'    => 0,
        'Text::TemplateLite'        => '0.01',
        'Text::Templet'             => '3.0',
        'Text::Tmpl'                => 0,
        #  0.2012 required to work on Strawberry Perl on Win32/XP
        'Text::Xslate'              => '0.2012',
        #  0.34 fixes (endian?) bug on Strawberry Perl on Win32/XP
        'Data::MessagePack'         => '0.34',
        #  1.0002 required for TextXlsateTT.
        'Text::Xslate::Bridge::TT2' => '1.0002',
        'Text::Xslate::Syntax::HTMLTemplate' => '0.1005',
    );

#  Prune the unsafe ones on windows.
delete @modules{ qw/
    HTML::Template::Compiled
    HTML::Template::JIT
    NTS::Template
    Template::Alloy
    Template::Tiny
    Text::ClearSilver
    Text::Template::Simple
    Text::Tmpl
    Term::ProgressBar::Simple
    / }
    if $^O =~ /MSWin/;

#  These appear to be fixed for windows.
#    Text::Clevery
#    Text::Xslate
#    Data::MessagePack
#    Text::Xslate::Bridge::TT2

my $builder = Module::Build->new(
    module_name         => 'Task::Template::Benchmark',
    license             => 'perl',
    dist_author         => q{Sam Graham <libtemplate-benchmark-perl@illusori.co.uk>},
    dist_version_from   => 'lib/Task/Template/Benchmark.pm',
    build_requires => {
        'Test::More'    => 0,
        #  0.36 required for "installdeps"
        'Module::Build' => '0.36',
    },
    requires => {
        %modules,
    },
    sign => 1,
    dynamic_config => 1,
);

$builder->create_build_script();
